test/SkipCallsTestHook.t.sol

```diff
@@ -0,0 +1,240 @@
+                            & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG
+                            & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG
+                            & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        MockERC20(Currency.unwrap(key.currency0)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+
+        assertEq(skipCallsTestHook.counter(), 0);
+        swapRouter.swap(key, swapParams, testSettings, abi.encode(address(this)));
+        assertEq(skipCallsTestHook.counter(), 1);
+    }
+
+    function test_beforeDonate_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.BEFORE_INITIALIZE_FLAG & ~Hooks.AFTER_INITIALIZE_FLAG & ~Hooks.BEFORE_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG
+                            & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG & ~Hooks.AFTER_SWAP_FLAG
+                            & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        assertEq(skipCallsTestHook.counter(), 0);
+        MockERC20(Currency.unwrap(key.currency0)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        MockERC20(Currency.unwrap(key.currency1)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        donateRouter.donate(key, 100, 200, abi.encode(address(this)));
+        assertEq(skipCallsTestHook.counter(), 1);
+    }
+
+    function test_afterDonate_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.BEFORE_INITIALIZE_FLAG & ~Hooks.AFTER_INITIALIZE_FLAG & ~Hooks.BEFORE_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG
+                            & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG & ~Hooks.AFTER_SWAP_FLAG
+                            & ~Hooks.BEFORE_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        assertEq(skipCallsTestHook.counter(), 0);
+        MockERC20(Currency.unwrap(key.currency0)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        MockERC20(Currency.unwrap(key.currency1)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        donateRouter.donate(key, 100, 200, abi.encode(address(this)));
+        assertEq(skipCallsTestHook.counter(), 1);
+    }
+}
```

Comment:
Could we add a test where we call again on the pool and the counter should be 2 but not 4?